### PR TITLE
Auth integration tests

### PR DESCRIPTION
**Pull request by kunagpal**
TODO: Streamline OAuth 2.0 flows.

### CODE REVIEW

File: test/integration/auth-methods/hawk.test.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+describe('Hawk auth', function () {
```
Comment (czardoz) [inline test/integration/auth-methods/hawk.test.js]: can you lowercase this (also for oauth1)?

File: test/integration/auth-methods/hawk.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        }, function (err, results) {
+            testrun = results;
+            done(err);
+        });
+    });
+
+    it('must have completed the run', function () {
+        expect(testrun).be.ok();
+        expect(testrun.done.calledOnce).be.ok();
+        testrun.done.getCall(0).args[0] && console.log(testrun.done.getCall(0).args[0].stack);
```
Comment (czardoz) [inline test/integration/auth-methods/hawk.test.js]: Use `console.error` or don't do this at all

File: test/integration/auth-methods/oauth1.test.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        }, function (err, results) {
+            testrun = results;
+            done(err);
+        });
+    });
+
+    it('must have completed the run', function () {
+        expect(testrun).be.ok();
+        expect(testrun.done.calledOnce).be.ok();
+        testrun.done.getCall(0).args[0] && console.log(testrun.done.getCall(0).args[0].stack);
```
Comment (czardoz) [inline test/integration/auth-methods/oauth1.test.js]: Use `console.error` or don't do this at all

Comment (kunagpal) [inline test/integration/auth-methods/oauth1.test.js]: Fixed

File: test/integration/auth-methods/hawk.test.js (hunk 2 above)
Comment (kunagpal) [inline test/integration/auth-methods/hawk.test.js]: Fixed

File: test/integration/auth-methods/hawk.test.js (hunk 1 above)
Comment (kunagpal) [inline test/integration/auth-methods/hawk.test.js]: Fixed

**Pull request merged**